folding/miners/folding_miner.py

```diff
@@ -242,18 +267,32 @@ def run(
                 file.write(content)
 
         for state, commands in commands.items():
-            bt.logging.info(f"Running {state} commands")
+            # Check to see if the simulation is taking longer than the timeout condition.
+            if (time.time() - start_time) > timeout:
```

Comment:
This won't interrupt a given step. I imagine that it's most likely that the current step is the production run when timeout is triggered. Two ways around this are to use [`Future.result(timeout=timeout)`](https://pythonhosted.org/futures/#concurrent.futures.Future.result) or manually specify the max time to GROMACS itself using [`-maxh`](https://manual.gromacs.org/current/user-guide/managing-simulations.html)